### ISSUE TITLE
Allow themes to specify number of columns in index

### ISSUE THREAD

**Issue by audiofanatic** [asciidoctor/asciidoctor-pdf#1663]
The number of columns in the index is currently hard-coded to 2. When a document contains lots of very long terms, there is a lot of wrapping and it isn't a great end result visually. The changes made by #645 have exacerbated the problem by introducing unwanted blank lines before wrapping such long entries and the added hanging indent actually makes the wrapping even worse. See screenshots of previous result using 1.5.0 (not sure of exact version) and now using the 1.5.3 release.

### asciidoctor-pdf 1.5.0 (or thereabouts):
<img width="299" alt="Index sample asciidoctor-pdf 1 5 0" src="https://user-images.githubusercontent.com/16160067/80301733-37b59c00-87e9-11ea-8563-561c651499ef.png">

### asciidoctor-pdf 1.5.3:
<img width="299" alt="Index sample asciidoctor-pdf 1 5 3" src="https://user-images.githubusercontent.com/16160067/80301744-400dd700-87e9-11ea-825f-ce5f2831fe07.png">

As a workaround for the above and maybe even a better permanent way to address it, being able to control the number of columns in the index would be great. I've done some local experimenting and it would seem fairly straightforward to enable themes to specify the number of columns. Please forgive the crude way of providing the code, I'm heavily focused on getting the published version of a book released in the next few days. Here's my slightly modified version of `convert_index_section` which lives in `converter.rb`:

```ruby
      def convert_index_section _node
        unless @index.empty?
          space_needed_for_category = @theme.description_list_term_spacing + (2 * (height_of_typeset_text 'A'))
          index_columns = @theme.index_columns || 2
          column_box [0, cursor], columns: index_columns, width: bounds.width, reflow_margins: true do
```

The above would be a backward-compatible change. It also gives weight to the broader suggestion of giving the index its own theme settings instead of re-using the description list theme details.

**Comment by mojavelinux** [comment]
>  The changes made by #645 have exacerbated the problem by introducing unwanted blank lines before wrapping

Eek! That wasn't supposed to happen.

> being able to control the number of columns in the index would be great.

Seems like a reasonable idea to me. Thanks for suggesting it.

**Comment by mojavelinux** [comment]
The first problem is now solved in master by #1665.
